### DISCUSSION TITLE
Failed to initialize vdo stream

### DISCUSSION THREAD

**Discussion by danudeep90** [AxisCommunications/acap-computer-vision-sdk-examples#137] [General] [answered]
Hi,

I am trying to run https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/tree/main/object-detector-python
application on AXIS camera P3268 LV Dome Camera. But, when i run the service, i am getting an error as "Failed to initialize vdo stream".

I suspect the error is in reading frames. I am unable to get any frames.

I even tried changing the videoCapture source in the code below to 0, but still doesnt work
```
cap = cv2.VideoCapture(1)
cap.set(cv2.CAP_PROP_FPS, 5)
cap.set(cv2.CAP_PROP_FRAME_WIDTH, 480)
cap.set(cv2.CAP_PROP_FRAME_HEIGHT, 320)
cap.set(cv2.CAP_PROP_FOURCC, cv2.VideoWriter_fourcc(*"RGB3"))
```

Can someone help me on this ?

**Comment by Corallo** [comment, marked as answer]
Hello @danudeep90 

Did you run the example 'as it is' following the instructions or did you make changes?
Did you use docker compose? 
In order for the container to access the video stream, it has to have some volumes that we set up in the [docker-compose file](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/blob/main/object-detector-python/docker-compose.yml#L16). Maybe you are starting the example without it?

**Reply by danudeep90** [reply]
Hi @Corallo ,
I am able to run the following docker command for static image and it works
`docker-compose --tlsverify --host tcp://$DEVICE_IP:$DOCKER_PORT --file static-image.yml --env-file ./config/env.$ARCH.$CHIP up`

But when i run the one for livestream, it doesn't work
`docker-compose --tlsverify --host tcp://$DEVICE_IP:$DOCKER_PORT --env-file ./config/env.$ARCH.$CHIP up
`
How do i setup those volumes ? is there a process ?

**Reply by danudeep90** [reply]
@Corallo : I ran the example as it is from the documentation.

**Reply by Corallo** [reply]
No, you don't have to do anything else than this, docker-compose looks for docker-compose file and creates the volumes for you.

If you didn't make any change to the code nor the docker files, maybe let's try to check the following
- Verify that your firmware version is up-to-date. Open the camera web gui and go to maintenance, and check if your firmware version is 11.3.
- Check in the log of docker-compose when you start the application, if it mentions some errors.

**Reply by danudeep90** [reply]
@Corallo : firmware version is 10.12.91. should i do an upgrade ?
no errors as such, frame is None, when i tried to print(frame) after line
` _, frame = cap.read()`

**Reply by Corallo** [reply]
Yes, the example is not compatible anymore with fw older than 11.0.
Either you update your firmware, (download the new one [here](https://www.axis.com/support/firmware))
or you use an [older version of the example](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/tree/v1.2) (not recommended).

**Reply by danudeep90** [reply]
I see an option on the GUI on maintenance section, firmware, I can use it to upgrade right ?

**Reply by Corallo** [reply]
Yes, the upgrade can be done using the GUI, follow the instructions there.

**Reply by danudeep90** [reply]
@Corallo : Thank you so much.. Firmware upgrade solved the problem !!
